services/QuillConnect/app/components/renderForQuestions/renderEndState.jsx

```diff
@@ -61,7 +61,7 @@ const EndState = React.createClass({
   renderTopThreeResponses() {
     let responses = this.props.responses;
     responses = hashToCollection(responses);
-    responses = responses.filter(response => response.optimal).sort((a, b) => b.count - a.count);
+    responses = responses.filter(response => response.optimal && response.text.length).sort((a, b) => b.count - a.count);
```

Comment:
It looks like we're covering for cases where empty strings have been marked as optimal, is that right?  If that's the case, then this feels like it might be a way of masking bad data that we should clean up in the response database rather than here, but I'm not totally sure.  Thoughts?

Reply:
@anathomical Yes, that is what that's doing. That was happening when admins added questions without adding an optimal response - I made a PR to fix that while I was doing this. Deleting any responses without text from the CMS database would also solve this problem, but I don't know a way to do that, do you?